lib/API/MP/v1/CustomFields.php

```diff
@@ -41,18 +49,14 @@ public function getSubscriberFields(): array {
       ],
     ];
 
-    $customFields = CustomField::selectMany(['id', 'name', 'type', 'params'])->findMany();
+    $customFields = $this->customFieldsRepository->findAll();
     foreach ($customFields as $customField) {
       $result = [
-        'id' => 'cf_' . $customField->id,
-        'name' => $customField->name,
-        'type' => $customField->type,
+        'id' => 'cf_' . $customField->getId(),
+        'name' => $customField->getName(),
+        'type' => $customField->getType(),
+        'params' => $customField->getParams(),
```

Comment:
Will $customField->getParams() handle both serialized and unserialized data correctly? I can see that it works for serialized data but I haven't checked unserialized data. Asking just to make sure that we don't need to treat the two cases differently as the old code did.

Reply:
I am not sure I understand correctly. The method `getParams` always returns `array` so I am not sure what to do here with unserialized data.

Reply:
Maybe there is nothing that we need to do here. I just wanted to make sure that you have checked that this should work for all cases as the old code had an if condition to handle the data differently if it was serialized or not (https://github.com/mailpoet/mailpoet/pull/3694/files#diff-0958c60d6aae5b820e6f9c13761826370dd48694a799c74dd98573ee0e214e7bL91). It is possible that Doctrine is handling this for us behind the scenes or something else. I have not checked.

Reply:
I know that the old model worked differently. But the database contains always a serialised value: https://d.pr/i/fk8Q0p and the entity always returns [array](https://github.com/mailpoet/mailpoet/blob/8cfd47e6568d330067d722a51c2524254e5fc5f7/lib/Entities/CustomFieldEntity.php#L62). SO I am not sure what to check.